Add `title` on the graph action elements for better a11y and user understanding

C-Loftus opened the pull request:
Currently the buttons on the site graph may be slightly confusing to new users if they are unfamiliar with the icons. By adding a `title` property on to the html element, it gives a popup tooltip on hover. 

This also makes it so a screen reader or other AT client could read the button and understand what it is doing. Currently since the buttons don't have a `title` or `arialabel` set, a screen reader would just read them as a generic `push button` which is inaccessible. 

I could not find a way to test this with an example so I may need guidance on how to test locally. 

```
host@computer ~/g/s/docs (a11yFixes)> npm i
npm error code EUNSUPPORTEDPROTOCOL
npm error Unsupported URL Type "workspace:": workspace:*
npm error A complete log of this run can be found in: /home/host/.npm/_logs/2025-09-07T18_47_12_197Z-debug-0.log
```

Fevol commented on packages/starlight-site-graph/components/graph/action-element.ts:
```diff
@@ -34,6 +35,7 @@ export function renderActionContainer(context: GraphComponent) {
 			actionElement.innerHTML = icons[('graph' + context.config.depth) as keyof typeof icons];
+			actionElement.title = `Depth: ${context.config.depth}`;
 			actionElement.onclick = e => {
 				context.config.depth = (context.config.depth + 1) % MAX_DEPTH;
```
The title should also be updated when the depth gets updated in the `onclick`

Fevol reviewed (commented):
Thanks for the PR! Everything looks good, if you want I can merge already (or I can wait if you want to fix the `depth` title update yourself)

Fevol commented:
As for running the docs, I am making use of [`bun`] as a package manager, which has the `workspace: *` for a more convenient monorepo setup. I'm not sure whether `npm` has something similar

C-Loftus commented:
> As for running the docs, I am making use of [`bun`](https://bun.com/) as a package manager, which has the `workspace: *` for a more convenient monorepo setup. I'm actually not entirely sure why `npm` doesn't work here.

Sounds good, I am about to start to the work week but I will find some time later in the week to get this installed and test locally I think

C-Loftus commented:
Ok I have tested this with bun and I can see the expected tooltip hover and also it reads correctly with my screen reader. So it should be all set.

I looked up proper semantics for how to name titles. From what I can see, a toggle button’s title/label describes the control itself (the feature), while its state indicates the current condition. Thus, I have changed the `title` attribute to be static since it just describes functionality, not state of the control. State would be properly described via `aria-pressed`. 

I have added both `aria-label` and `title` since some screen readers don't read title. There is essentially no downside to putting this in with the same info.

Fevol commented:
I'm really sorry for the long wait. Things have been busy on my site (I started working, and didn't have a lot of spare time to focus on these projects.)

Many thanks for the addition! (And no worries about the diff!)
